### PR TITLE
FOUR-20467 implement a caching system for settings

### PR DESCRIPTION
**Pull request by devmiguelangel**
## Issue & Reproduction Steps
Implement a Caching System for Settings

## Solution
- Add the `CacheInterface` interface
- Add the `SettingCacheManager` class 
- Add the `SettingCacheFacade` facade
- Register the facade in `ProcessMakerServiceProvider.php`

## Related Tickets & Packages
[FOUR-20467](https://processmaker.atlassian.net/browse/FOUR-20467)

## Code Review Checklist
- [ ] I have pulled this code locally and tested it on my instance, along with any associated packages.
- [ ] This code adheres to [ProcessMaker Coding Guidelines](https://github.com/ProcessMaker/processmaker/wiki/Coding-Guidelines).
- [ ] This code includes a unit test or an E2E test that tests its functionality, or is covered by an existing test.
- [ ] This solution fixes the bug reported in the original ticket.
- [ ] This solution does not alter the expected output of a component in a way that would break existing Processes.
- [ ] This solution does not implement any breaking changes that would invalidate documentation or cause existing Processes to fail.
- [ ] This solution has been tested with enterprise packages that rely on its functionality and does not introduce bugs in those packages.
- [ ] This code does not duplicate functionality that already exists in the framework or in ProcessMaker.
- [ ] This ticket conforms to the PRD associated with this part of ProcessMaker.

[FOUR-20467]: https://processmaker.atlassian.net/browse/FOUR-20467?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: ProcessMaker/Cache/Settings/SettingCacheManager.php [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+     * Get a value from the settings cache.
+     *
+     * @param string $key
+     * @param mixed $default
+     *
+     * @return mixed
+     */
+    public function get(string $key, mixed $default = null): mixed
+    {
+        return $this->cacheManager->get($key, $default);
```
Comment (rodriquelca) [inline ProcessMaker/Cache/Settings/SettingCacheManager.php:39]: improve with error handling example
   ```
try {
    return $this->cache->get($key, $default);
       } catch (\Exception $e) {
           // Log the error
           \Log::error("Cache get error: " . $e->getMessage());
           return $default;
       }
```

File: ProcessMaker/Providers/ProcessMakerServiceProvider.php [hunk 2]
```diff
@@ -164,6 +165,10 @@ public function register(): void
         $this->app->singleton('compiledscreen', function ($app) {
             return new ScreenCompiledManager();
         });
+
+        $this->app->singleton('setting.cache', function ($app) {
```
Comment (rodriquelca) [inline ProcessMaker/Providers/ProcessMakerServiceProvider.php:169]: maybe we can validate if cache configuration is enabled ana all necessary settings are present and correct. before to instance to settings cache managger
example

```
public function boot() {
           $this->validateCacheConfig();
       }

       protected function validateCacheConfig() {
           if (!config('cache.default')) {
               throw new \Exception('Default cache configuration is missing.');
           }
       }
```

File: ProcessMaker/Cache/Settings/SettingCacheManager.php (hunk 1 above)
Comment (devmiguelangel) [reply ProcessMaker/Cache/Settings/SettingCacheManager.php]: Error handling was added

File: ProcessMaker/Providers/ProcessMakerServiceProvider.php (hunk 2 above)
Comment (devmiguelangel) [reply ProcessMaker/Providers/ProcessMakerServiceProvider.php:170]: Validation was added

**Review by rodriquelca — APPROVED** [review]

File: ProcessMaker/Cache/Settings/SettingCacheManager.php (hunk 1 above)
Comment (devmiguelangel) [reply ProcessMaker/Cache/Settings/SettingCacheManager.php]: By the way, there is a ticket [FOUR-20909](https://processmaker.atlassian.net/browse/FOUR-20909) to go deeper into it.

**Pull request merged**